### PR TITLE
add support for different regional language in contentV3 and IVR

### PR DESCRIPTION
**Pull request by farmanahmed888**
This pull request updates language support across both the web application and IVR backend. The main changes involve replacing Marathi with Odia in language selection components, and adding Odia, Hindi, and Tamil to the list of supported languages for IVR dialog URLs. Additionally, the backend now ensures that only pull-model, non-deleted content is fetched, improving content filtering.

**Language support updates**

* Replaced Marathi with Odia in the language selection dropdowns for both `AddQuiz` and `AddStory` components in the web app (`ContentWebApp/src/components/AddQuiz.js`, `ContentWebApp/src/components/AddStory.js`). [[1]](diffhunk://#diff-56d90a051f35d89adefe798ca83802bfb4c57b83d13a228e2e31d0754d89e058L161-R161) [[2]](diffhunk://#diff-d327286785b1517f46172085a9bd3be4764e6d1133b4f188f1025f66d7cd07a8L436-R436)
* Added Odia, Hindi, and Tamil to the `languageDialogUrls` dictionary in both `IVRv2/app/fsm/url_paths.py` and `IVRv2/app/fsm/ivr_constants.py`, enabling IVR dialog support for these languages. [[1]](diffhunk://#diff-5fe0d262b470210a2fec288f87972e2dc96cea540dbb2bf89c2cf43d9ee5adddL15-R18) [[2]](diffhunk://#diff-4ac0f2653fcb953f33e64088ba9b81dae217a7baffcc3a200d9224841f2c1aa2L17-R20)

**Backend content filtering improvements**

* Modified `instantiate_from_latest_content` in `IVRv2/app/fsm/insti.py` to fetch only pull-model, non-deleted content, and optionally restrict to specific content IDs if provided. This ensures better filtering of content for the FSM.

### CODE REVIEW

**Review by a4i-architect — COMMENTED** [review]
### Review Summary

**Key Changes & Observations:**

1. **UI Language Options Updates:**
   - Correctly replaced "Marathi" with "Odia" in dropdowns for both AddQuiz and AddStory components.
   - Updated language dialog URLs to include `hindi`, `tamil`, and `odia` in constants files (`ivr_constants.py`, `url_paths.py`).
   - These updates are consistent and maintain backward compatibility.

2. **Code Quality & Consistency:**
   - No evident issues with the React components; the dropdown updates are straightforward.
   - The URL path updates are uniform across files.

3. **Backend Content Retrieval:**
   - Improved the `instantiate_from_latest_content` function to filter for `isPullModel` and non-deleted items, optionally restricting by `content_ids`.
   - This enhances query specificity and data integrity.

**Issues & Recommendations:**

- *Minor:* The code comments could be more detailed for clarity (e.g., specify why these particular filters are used).
- *No detection of ill-typed or duplicated code*.
- *No apparent cyclomatic complexity concerns*, the query logic is simple.

### Overall Assessment & Score

- The updates are straightforward, correct, and improve language coverage.
- Minor improvements can be made in commenting and maybe centralizing language options to avoid duplication.

**Final Score:** **85/100**   
(Approved with minor suggestions for code comments and possibly centralizing language configurations to prevent repetition.)

**Review by a4i-architect — COMMENTED** [review]
### High-Level Summary & Key Issues

The changes primarily involve:
- Updating language options from "marathi" to "odia" across multiple UI components.
- Extending language support in backend constants and URLs to include Hindi, Tamil, and Odia, aligning with the UI updates.
- Modifying the database query in the backend to filter for `isPullModel` and `!isDeleted`, and allowing optional filtering by `content_ids`.
- Mock test class expanded with a dummy `query_items` method returning all data, likely for testing purposes.

### Issues & Recommendations

1. **Inconsistent Language Options**  
   - UI components previously had Marathi; now replaced with Odia. Ensure this change is intentional and synchronized with backend options & resources.
   - Update English comments/labels if any mention Marathi to Odia to avoid confusion.

2. **Backend Query Logic**  
   - The new query filters `isPullModel` and `!isDeleted`, which is good.  
   - But the dummy `query_items` method in tests always returns all content; this bypasses actual filtering logic—acceptable for tests but should be noted.

3. **Code Quality & Typing**  
   - Typings in backend (`query_items`) are inferred but not explicit. Consider adding type hints for clarity.
   - Ensure the `content_ids` variable in the backend is properly validated and defaults are set if missing.

4. **Design & Complexity**  
   - The backend change is straightforward; no overcomplexity detected.  
   - The UI updates are simple replacements, with no notable technical concerns.

5. **Testing & Coverage**  
   - The mock `query_items` always returning `find_all()` might mask filtering bugs. Extend tests to verify filtering behavior explicitly.
   
### Overall Score: 80/100

- **Strengths:** Clear updates to support additional languages, aligned backend filtering logic.
- **Weaknesses:** Mock method oversimplifies, some comments could clarify intent.
- **Acceptance:** With minor improvements and better test coverage, the patch is ready for milestones.

File: IVRv2/tests/test_insti.py [hunk 1]
```diff
@@ -42,6 +42,9 @@ async def find_all(self):
             }
         ]
 
+    async def query_items(self, query):
+        return await self.find_all()
```
Comment (theainerd) [inline IVRv2/tests/test_insti.py]: **Test mock `query_items` ignores the query filter — test does not validate the new filtering logic**

The `query_items` stub simply delegates to `find_all()` and ignores the `query` parameter. Meanwhile, the fixture data at line 37 has `"isPullModel": False`. In production, the new query in `insti.py` filters for `{"isPullModel": True, "isDeleted": {"$ne": True}}`, so this document would be excluded. The test passes only because the mock doesn't apply the filter. Consider updating the fixture data to have `"isPullModel": True` or implementing actual filter logic in the mock so the test meaningfully exercises the new code path.

File: IVRv2/app/fsm/url_paths.py [hunk 2]
```diff
@@ -12,7 +12,10 @@
 languageDialogUrls = {
   'english':'languageDialog/english/For%20English/{speechRate}.mp3',
   'kannada':'languageDialog/kannada/For%20Kannada/{speechRate}.mp3',
-  'bengali':'languageDialog/bengali/For%20Bengali/{speechRate}.mp3'
+  'bengali':'languageDialog/bengali/For%20Bengali/{speechRate}.mp3',
+  'hindi':'languageDialog/hindi/For%20Hindi/{speechRate}.mp3',
+  'tamil':'languageDialog/tamil/For%20Tamil/{speechRate}.mp3',
+  'odia':'languageDialog/odia/For%20Odia/{speechRate}.mp3',
 }
```
Comment (theainerd) [inline IVRv2/app/fsm/url_paths.py]: **Duplicated `languageDialogUrls` dict in both `ivr_constants.py` and `url_paths.py`**

The `languageDialogUrls` dictionary is now maintained identically in two separate files. Any future language additions must be made in two places, which is error-prone. Consider consolidating into a single location and importing from there.

File: ContentWebApp/src/components/AddQuiz.js [hunk 3]
```diff
@@ -158,7 +158,7 @@ const AddQuiz = ({ quiz }) => {
             >
               <option value="kannada">Kannada</option>
               <option value="hindi">Hindi</option>
-              <option value="marathi">Marathi</option>
+              <option value="odia">Odia</option>
```
Comment (theainerd) [inline ContentWebApp/src/components/AddQuiz.js:162]: **Marathi is removed rather than Odia being added alongside it**

Are we removing Marathi support? If any existing content is tagged with `"marathi"`, users will no longer be able to select that language in the UI to view or edit it.

**Review by theainerd — COMMENTED** [review]
Review comments on PR #155

File: ContentWebApp/src/components/AddQuiz.js (hunk 3 above)
Comment (farmanahmed888) [reply ContentWebApp/src/components/AddQuiz.js:162]: > Are we removing Marathi support? If any existing content is tagged with "marathi", users will no longer be able to select that language in the UI to view or edit it.


Previously there exists no support for marathi, it was just a filler, same as Tamil and hindi, they also did not have any support in IVR v2,

but currently I have updated the blob to support hindi, tamil, and odia, but not marathi, hence the change

**Review by a4i-architect — COMMENTED** [review]
**Summary & Key Observations:**

- **Simple UI text updates**: Replacing 'marathi' with 'odia' in select options is straightforward and correct.
- **Code consistency**: The change in `AddStory.js` mirrors the previous one in `AddQuiz.js`; consistent and correct.
- **Backend content fetch**: The modification in `insti.py` to filter only pull-model, not-deleted content with optional `content_ids` is a solid improvement:
  - Properly refines queries, avoiding unnecessary data transfer.
  - Uses `query_items()` method, which is mock-implemented correctly.
  - Minor note: Enhance error handling or ensure `content_ids` is always provided/validated.
- **Constants update**: Adding more languages to `ivr_constants.py` and referencing it in `url_paths.py` is good.  
  - Issue: In `url_paths.py`, the `languageDialogUrls` is redefined via import from `ivr_constants`; ensure no conflicts.  
  - Also, the previous duplicate dictionary in `url_paths.py` is removed correctly.
- **Test updates**: Mock `query_items()` in `test_insti.py` is well-implemented:
  - Correctly filters content based on query operators.
  - Consider adding more complex queries or edge cases.
  
**Issues & Recommendations:**

- **Hardcoded language options**: Replacing 'marathi' with 'odia' in UI selects suggests localization intent; ensure all language options are consistent across components and backend constants.
- **Potential inconsistency**:  
  - In `IVRv2/app/fsm/url_paths.py`, importing `languageDialogUrls` from `ivr_constants.py`.  
  - Confirm that all language keys are synchronized, and no duplicate definitions exist.
- **Type safety & validation**:  
  - In `query_items()`, assume query structure; consider more robust operator support (e.g., `$in`, `$eq`) for production.
- **Testing & Mocking**:  
  - Ensure test coverage for new filtering logic, especially with multiple conditions.
  - Maintain clear separation between mock data and production code.

**Design & Complexity:**

- The code is straightforward, with clear separation of UI and backend logic.
- Mocking in tests is simple and effective.
- No evident over-complication or cyclomatic complexity issues.

**Score: 85/100**

- **Accept** with minor improvements: validation, comprehensive test coverage, and ensuring consistent language options across all components.

File: IVRv2/app/fsm/url_paths.py (hunk 2 above)
Comment (farmanahmed888) [reply IVRv2/app/fsm/url_paths.py]: done

File: IVRv2/tests/test_insti.py (hunk 1 above)
Comment (farmanahmed888) [reply IVRv2/tests/test_insti.py]: done

File: ContentWebApp/src/components/AddQuiz.js (hunk 3 above)
Comment (Soumabha) [inline ContentWebApp/src/components/AddQuiz.js]: Why replace Marathi? Lets have it as an extra option.

**Review by a4i-architect — COMMENTED** [review]
**Overall Summary:**

The modifications mainly involve replacing "Marathi" with "Odia" across UI components, language constants, and backend code, aligning with new language support. Additional changes include improving content retrieval logic in `insti.py` with filtering by pull-model and deletion status, as well as updating URL path and TTS configuration to include "Odia."

**Key Issues & Recommendations:**

1. **UI Changes (AddQuiz, AddStory):**  
   - Correct and consistent language options seem maintained; no issues here.

2. **Backend (`insti.py`):**  
   - The `query_items()` method in the mock is well-implemented. However, this method is added only for testing; in production, ensure proper database querying (e.g., MongoDB query syntax).  
   - Be cautious with the `content_ids` filter; validate that `content_ids` exists and is correctly passed.

3. **Constants and URL Config:**
   - The inclusion of "odia" in `languageDialogUrls` and exported constants is correct.  
   - Import redundancy in `url_paths.py`—since `languageDialogUrls` is imported from another module, defining it again locally is unnecessary; remove duplicate definitions to avoid confusion.

4. **TTS Service Config:**
   - Adding "or" (Odia) voice code is correct, but verify that the voice name `"or-IN-SubhasiniNeural"` exists in your TTS provider.  
   - Maintain consistent naming conventions.

5. **Code Quality & Testing:**
   - The mock `query_items()` is simplified; ensure your production code uses efficient database querying mechanisms.
   - Tests are updated, but consider adding unit tests for the `query_items()` method.

6. **Design & Complexity:**
   - Logic is straightforward, no notable complexity issues.
   - Maintain clear separation of configuration data and logic.

**Acceptance Score: 75/100**  
The core changes are appropriate, but some cleanup (removing redundancies, confirming resource existence) and ensuring production-ready queries are necessary before full deployment.

**Review by a4i-architect — COMMENTED** [review]
**High-Level Summary & Analysis:**

**Key Changes & Intentions:**
- Added support for the "Odia" language across multiple components (UI options, constants, storage URLs, and TTS configs).
- Modified data fetching in `IVRv2/app/fsm/insti.py` to filter `isPullModel` and `isDeleted` content, with improved query logic.
- Extended `languageDialogUrls` to include Odia and other languages for playback.
- Updated tests to reflect new language and to mock the `query_items` method.
- Removed redundant `languageDialogUrls` declaration from `url_paths.py`.

---

**Issues & Recommendations:**

1. **Code Redundancy & Duplication:**
   - `languageDialogUrls` is duplicated: once in `IVRv2/app/fsm/ivr_constants.py` and once (commented out) in `url_paths.py`. Remove the outdated or commented version to prevent confusion.
   
2. **Type & Consistency Checks:**
   - The new `<option>` tags are straightforward, but ensure frontend validation if user inputs rely on these options.
   - Adding "Odia" as a language appears correct, but verify no existing dependencies break with this addition.

3. **Query Logic in `IVRv2/app/fsm/insti.py`:**
   - The `query_items` mock manages simple `$ne` and `$in` queries properly; however, support for other operators or nested queries may be necessary for production. Currently, it's adequate for tests.

4. **Test Coverage & Mocking:**
   - The mock `query_items` method in tests is well-designed for `$ne` and `$in`. Ensure tests explicitly verify retrieval of Odia content where relevant.

5. **Design & Patterns:**
   - The approach to add new language support is mainly data-driven (constants, UI options, configs). This is flexible and scalable, fitting best practices for language localization.
   - No complexity issues detected; the logic is simple.

6. **Version & Deployment:**
   - Confirm all references and dependent files are updated simultaneously to prevent inconsistencies.

---

**Overall Score: 85/100**  
- Sufficient localization extension
- Minor cleanup needed (remove duplicate configs)
- Adequate mock for testing
- Consider future-proofing for more operators in mocks

**Acceptance is recommended once cleanup is done and testing confirms Odia support end-to-end.**

**Review by theainerd — APPROVED** [review]
The patch correctly adds Odia, Hindi, Marathi and Tamil language support across ContentWebApp UI dropdowns, IVR constants, and TTS service. All three prior review comments have been properly addressed in the latest commits. The remaining findings are low-priority cleanup items that do not affect functionality

**Review by a4i-architect — COMMENTED** [review]
**High-Level Review and Feedback:**

1. **Coverage Workflow Enhancements**:
   - The added step lists artifacts, which improves traceability.
   - Removing the dependency installation step reduces unnecessary overhead; ensure coverage tools are pre-installed or handled elsewhere.

2. **UI Component Updates**:
   - Adding "Odia" language option in quiz and story components is straightforward; confirm consistency across all language lists.
   - No issues here; just ensure front-end test coverage.

3. **Backend Data Fetching**:
   - The `instantiate_from_latest_content` function now filters for pull-model, non-deleted content, which is good.
   - However, the custom `query_items` in tests is simplified; for accurate mock testing, consider using a more robust in-memory query filter or a mock library to reflect real query behavior.

4. **Constants & Path Files**:
   - Synchronization of language dialog URLs between `ivr_constants.py` and `url_paths.py` is improved; ensure the URL format is consistently used in actual playback.
   - Removing duplicate `languageDialogUrls` from `url_paths.py` is good; avoid redefining variables.

5. **Test Coverage & Mocking**:
   - The mock `query_items` is rudimentary. For robustness, consider more comprehensive query support.
   - No automated tests or assertions are shown; ensure proper unit tests exist for new filters and language options.

6. **Code Style & Quality Issues**:
   - No major issues detected.
   - Minor: Remove commented-out or redundant code if not going to be used (e.g., in `IVRv2/app/fsm/insti.py`).

7. **Design & Complexity**:
   - Logic appears straightforward; no over-complexity detected.
   - The filtering logic for content in tests could be simplified, but it's acceptable for mock purposes.

**Score**: 80/100  
*Recommend acceptance after minor improvements in test mocks and ensuring consistent URL usage.*

**Review by a4i-architect — COMMENTED** [review]
**High-Level Summary & Review**

The commits aim to extend language support across multiple components, especially adding "Odia" (odia). Key areas updated include UI options (`AddQuiz.js`, `AddStory.js`), content fetching (`insti.py`), constants (`ivr_constants.py`), URL path configs (`url_paths.py`), mock data for tests (`test_insti.py`), and TTS integrations (`ttsService.js`).  

**Key Issues & Recommendations**

1. **Redundancy & Inconsistency:**
   - The language options are duplicated across UI, constants, and backend configs. To avoid divergence, centralize language support mappings in a single module or configuration file.
   
2. **Mock Data (test_insti.py):**
   - The `query_items` method is added for testing purpose but is overly simplified. It could be more robust or replaced with proper test fixtures if available.
   
3. **Code Quality Concerns:**
   - The `IVRv2/app/fsm/insti.py` update modifies the query to filter for pull model and not-deleted content, which is good. However, avoid assuming such methods (`query_items`) exist; this is a mock adaptation and should reflect actual database interfaces.
   
4. **Missing Error Handling & Edge Cases:**
   - When adding new languages, ensure fallback or error handling for unsupported language codes in all integrative components.
   
5. **Design & Modularity:**
   - Instead of spreading language support across multiple files, create a unified `languages.py` module containing all mappings (display names, URL templates, TTS codes, etc.). This improves maintainability.
   
6. **Unnecessary Changes:**
   - The removal in `url_paths.py` of the inline `languageDialogUrls` is correct; however, it assumes that the import from `ivr_constants` covers all needs. Confirm no other parts rely on the removed local variable.
   
7. **Typings & Consistency:**
   - Verify that all language codes (e.g., `'odia'`) are used consistently across all modules, including UI, backend, constants, and external services.

**Overall Score: 78/100**

*Acceptable with recommendations.*  
- Make the language support a centralized module.  
- Ensure test coverage includes new languages.  
- Confirm TTS, URL, and dialog configurations are synchronized with the central source.  
- Remove or improve mock implementations for production readiness.  

**References:**  
- Modular configuration best practices: [12 Factors App - Config](https://12factor.net/config)  
- Mock data management: [Testing in React/Node with Mocks](https://jestjs.io/docs/manual-mocks)  
- Internationalization best practices: [i18n Strategy](https://www.i18next.com/overview/configuration-overview)

File: IVRv2/tests/test_insti.py [hunk 4]
```diff
@@ -34,14 +34,37 @@ async def find_all(self):
...
+    async def query_items(self, query):
+        all_items = await self.find_all()
+        if not query:
+            return all_items
+
+        def matches(doc, query_dict):
+            for key, cond in query_dict.items():
+                # simple operator object (e.g. {"isDeleted": {"$ne": True}})
+                if isinstance(cond, dict):
+                    if "$ne" in cond:
```
Comment (N1kunj1998) [inline IVRv2/tests/test_insti.py]: We should also add support for `$in` in matches so that it supports this type of query `{"_id": {"$in": content_ids}, "isPullModel": True, "isDeleted": {"$ne": True}}`

```
if "$in" in cond:
    if doc.get(key) not in cond["$in"]:
        return False
elif "$ne" in cond:
```

File: IVRv2/tests/test_insti.py
Comment (N1kunj1998) [inline IVRv2/tests/test_insti.py:17]: This test only checks for a document with `isPullModel: True` and `isDeleted: False`. There is no test that:
- Excludes documents with `isPullModel: False`
- Excludes documents with `isDeleted: True`
- Asserts the `content_ids` path returns only matching IDs

If someone later removes the filter and reverts to find_all(), all content (including deleted and non–pull-model) would be used again. The existing test would still pass, because it only ever had one “good” document. So the test does not protect the new behavior.

we need to add more tests like this:

```
@pytest.mark.asyncio
async def test_deleted_content_is_excluded():
    """Documents with isDeleted=True must not appear in the FSM."""
    ...

@pytest.mark.asyncio
async def test_non_pull_model_content_is_excluded():
    """Documents with isPullModel=False must not appear in the FSM."""
    ...
```

Comment (N1kunj1998) [inline IVRv2/tests/test_insti.py]: `IVRv2/tests/mocks/mock_database.py` in this file also we are having mock database, why are we having two different in-memory mock implementations?

**Comment by N1kunj1998** [comment]
[#232](https://github.com/A4i-tech/.github/issues/232) - issue mentions we should be able to add only tamil details without english and it should work but it does not, not sure if real issue or Issue description is not correct.

image

**Comment by farmanahmed888** [comment]
> `IVRv2/tests/mocks/mock_database.py` in this file also we are having mock database, why are we having two different in-memory mock implementations?

we require english title

**Review by a4i-architect — COMMENTED** [review]
**Summary of Review**:

The updates focus on adding support for Odia language across frontend options, backend content filtering, IVR constants, URL paths, and TTS configurations, plus extensive mock data improvements for testing. The pattern of adding a new language is consistent but somewhat repetitive.

**Key issues & suggestions**:

1. **Code Duplication & Manual Updates**:
   - Repeated language options in `AddQuiz.js` and `AddStory.js`.
   - Consider centralizing supported languages in a single configuration file to reduce redundancy and maintenance effort.

2. **Content Filtering Logic Consistency**:
   - Mock data now includes diverse scenarios (deleted, non-pull, valid), which is good.
   - The `query_items()` method's operator support is comprehensive but may lack edge cases (e.g., `$and`, `$or`). For current tests, it suffices.

3. **Test Quality & Coverage**:
   - Tests now robustly verify filtering logic, especially exclusion of deleted and non-pull content.
   - Additional test coverage could include invalid query operators or malformed data.

4. **Design & Architecture**:
   - The code design for filtering is straightforward; centralizing language options and flags could streamline.
   - Mock data is well-structured, but for scalability consider using parametrized test cases.

5. **Minor Issues**:
   - Ensure all language options (e.g., 'odia') are reflected uniformly where relevant.
   - Maintain consistency in naming conventions (e.g., `content_ids` vs `content_ids`).

**Overall Score**: **75/100**  
*Rationale*: Correctly implements language expansion and robust testing; improvements needed in centralization and potential operator support in queries.

**Acceptance recommendation**: **Conditional** — pending refactoring of language options into a shared config and validation of query operator coverage if needed.

File: IVRv2/tests/test_insti.py
Comment (farmanahmed888) [reply IVRv2/tests/test_insti.py]: done

Comment (farmanahmed888) [reply IVRv2/tests/test_insti.py:17]: done

File: IVRv2/tests/test_insti.py (hunk 4 above)
Comment (farmanahmed888) [reply IVRv2/tests/test_insti.py]: done

File: IVRv2/app/fsm/url_paths.py [hunk 5]
```diff
@@ -1,4 +1,6 @@
 from app.settings import settings
+from app.fsm.ivr_constants import languageDialogUrls
```
Comment (Soumabha) [inline IVRv2/app/fsm/url_paths.py]: Unused, please remove.

File: IVRv2/app/fsm/url_paths.py [hunk 6]
```diff
@@ -9,12 +11,6 @@
     'authToken': 'postman'
 }
 
-languageDialogUrls = {
-  'english':'languageDialog/english/For%20English/{speechRate}.mp3',
-  'kannada':'languageDialog/kannada/For%20Kannada/{speechRate}.mp3',
-  'bengali':'languageDialog/bengali/For%20Bengali/{speechRate}.mp3'
-}
-
 speechRate = "1.0"
```
Comment (Soumabha) [inline IVRv2/app/fsm/url_paths.py]: Are we using the speechRate from url_paths.py or from ivr_constants.py? Lets keep only single sources of truths to make modifications and extensions easier.

Do the same for `readingContentTitlesDialogUrl` variable and others too as applicable with copies in the `ivr_constants.py`

**Review by Soumabha — APPROVED** [review]

**Review by N1kunj1998 — APPROVED** [review]
Changes looks good

File: IVRv2/app/fsm/url_paths.py (hunk 5 above)
Comment (farmanahmed888) [reply IVRv2/app/fsm/url_paths.py]: the file is useless removed it

File: IVRv2/app/fsm/url_paths.py (hunk 6 above)
Comment (farmanahmed888) [reply IVRv2/app/fsm/url_paths.py]: the file is useless removed it

**Review by a4i-architect — COMMENTED** [review]
### High-Level Review Summary

The code introduces new language options (Odia, Hindi, Tamil, Marathi) in content options, constants, and URLs, which enhances multilingual support. Modifications to data retrieval now filter content with `isPullModel=True` and `isDeleted!=True`. 

### Issues & Recommendations

1. **AddQuiz.js / AddStory.js**: 
   - Correctly added `<option>` for "odia"; no issues here.

2. **Content Filtering Logic (IVRv2/app/fsm/insti.py)**:
   - The query logic now explicitly filters on `isPullModel` and `isDeleted`. Good.
   - Ensure the `query_items` method supports operators; mock implementation has a basic but adequate support.
   - No direct bugs, but consider validating the data with comprehensive tests for edge cases.

3. **MockDatabase Enhancements**:
   - The `query_items` method now supports `$ne`, `$in`, `$gt`, etc., which aligns with real MongoDB. This is good.
   - Test data covers various content states, including deleted and non-pull models, which is excellent.
   - No apparent ill-typed code, but keep the logic in `matches()` concise to avoid bugs.

4. **Tests (IVRv2/tests/test_insti.py)**:
   - Testing for inclusion/exclusion based on flags is solid.
   - Maintain the test data as representative—good practice.
   - Additional tests verifying the correctness of filtering queries and edge cases are recommended.

5. **Constants Files (IVRv2/app/fsm/ivr_constants.py)**:
   - Added language URLs, dialog URLs, and voice names support new languages properly.
   - Check for duplicate entries or inconsistent data if more languages are added in future.

6. **URL Paths (IVRv2/app/fsm/url_paths.py)**:
   - The entire URL URLPaths file is commented out, which might be intentional. Ensure this is not accidental.
   - If still needed, port the URL patterns into the constants files with support for new languages.

7. **Service Layer (backend-server)**:
   - Added `or` language code for Odia in TTS, which aligns with URL support.
   - No issues identified; ensure appropriate usage in the application.

### Design & Complexity
- Content filtering via explicit flags is straightforward. 
- Using operator-aware matching in mocks improves test reliability.
- Questionable if the mocks should include more complex nested data or simulate real responses.

### Final Score: **85/100**
- The implementation adds valuable multilingual support and refines content filtering.
- Minor improvements: verify URL patterns are active, expand test coverage for edge cases, and ensure no accidental comment blocks.

**Overall**: Accept with minor suggestions to improve URL pattern management and broaden test coverage.
